tools/shellcheck-all.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+set -euo pipefail
+
+cd "$(git rev-parse --show-toplevel)"
+
+failed=""
+while IFS="" read -r file
+do
+    if grep -Eq '^#!(.*/|.*env +)(sh|bash|ksh)' "$file" || (echo "$file" | grep -Eq '.*.(ba)?sh$')
```

Comment:
I believe the `echo | grep` here can be replaced with a usage of the regex matching operator?

Reply:
TIL